syntax = "proto3";

import "google/protobuf/timestamp.proto";

service TaskManager { 
   rpc CreateTask(Task) returns (TaskResponse) {}
   rpc UpdateTask(TaskWithId) returns (TaskResponse) {}
   rpc DeleteTask(TaskWithId) returns (TaskResponse) {}
   rpc GetMyTasks(TaskWithId) returns (Tasks) {}
   rpc GetTaskById(TaskWithId) returns (Task) {}
}

message Task {
    uint64 userId = 1;
    string content = 2;
    google.protobuf.Timestamp dateOfCreation = 3;
    google.protobuf.Timestamp deadline = 4;
    uint64 status = 5;
}

message Tasks {
    repeated Task tasks = 1;
}

message TaskWithId {
    uint64 id = 1;
    Task task = 2;
}

message TaskWithIds {
    repeated TaskWithId tasks = 1;
}

message TaskResponse {
    uint64 id = 1;
    string status = 2;
}